nanome_docking/menus/DockingMenu.py

```diff
@@ -105,438 +122,358 @@ def refresh_run_btn_unusable(self, update=True, after=False):
-        self._run_button = run_button
-        self._run_button.enabled = False
-        self.refresh_run_btn_unusable()
+    def multi_select_dropdown(self, dropdown, item):
+        if not hasattr(dropdown, '_selected_items'):
+            dropdown._selected_items = []
 
-        pose_sub_btn = menu.root.find_node("PoseSub").get_content()
-        pose_sub_btn.register_pressed_callback(pose_subbed_callback)
-        pose_add_btn = menu.root.find_node("PoseAdd").get_content()
-        pose_add_btn.register_pressed_callback(pose_added_callback)
+        selected_items = dropdown._selected_items
+        if item not in selected_items:
+            selected_items.append(item)
+        else:
+            selected_items.remove(item)
+            item.selected = False
 
-        location_refresh_btn = menu.root.find_node("LocationRefresh").get_content()
-        location_refresh_btn.register_pressed_callback(loc_refresh_pressed_callback)
+        for ddi in selected_items:
+            ddi.selected = True
 
-        # loading bar
-        self.ln_loading_bar = menu.root.find_node("LoadingBar")
-        self.loading_bar = self.ln_loading_bar.get_content()
-        self.loading_bar.description = "    Loading...          "
+        dropdown.use_permanent_title = True
+        permanent_title = ','.join([ddi.name for ddi in selected_items]) if selected_items else 'None'
+        dropdown.permanent_title = permanent_title
 
-        # lists
-        self._complex_list = menu.root.find_node("ComplexList").get_content()
-        self._score_list = menu.root.find_node("ScoreList").get_content()
-        self._receptor_tab = menu.root.find_node("ReceptorButton").get_content()
-        self._ligand_tab = menu.root.find_node("LigandButton").get_content()
-        self._site_tab = menu.root.find_node("SiteButton").get_content()
+        dropdown.use_permanent_title = len(selected_items) > 1
+        self._plugin.update_content(dropdown)
 
-        # dropdown
-        self._ligand_dropdown = menu.root.find_node("LigandDropdown").add_new_dropdown()
-        self._ligand_dropdown.use_permanent_title = True
-        self._ligand_dropdown.permanent_title = "None"
-        self._receptor_dropdown = menu.root.find_node("ReceptorDropdown").add_new_dropdown()
-        self._receptor_dropdown.use_permanent_title = True
-        self._receptor_dropdown.permanent_title = "None"
-        self._site_dropdown = menu._root.find_node("SiteDropdown").add_new_dropdown()
-        self._site_dropdown.use_permanent_title = True
-        self._site_dropdown.permanent_title = "None"
 
-        # slider
-        self._slider = menu.root.find_node("Slider").get_content()
-        self._slider.register_released_callback(slider_released_callback)
-        self._slider.current_value = self._autobox
+class SettingsMenu:
+
+    def __init__(self, plugin):
+        self._plugin = plugin
+        self._menu = nanome.ui.Menu.io.from_json(os.path.join(BASE_DIR, 'jsons', 'docking_settings.json'))
```

Comment:
Nbd, but I prefer the `os.path.join...` to be a const at top of file, `SETTINGS_MENU_PATH` and same for `DOCKING_MENU_PATH` for the other one